dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNRatioTapChangerInterfaceIIDM.cpp

```diff
@@ -0,0 +1,146 @@
+//
+// Copyright (c) 2015-2020, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+// This file is part of Dynawo, an hybrid C++/Modelica open source time domain
+// simulation tool for power systems.
+//
+
+/**
+ * @file  DYNRatioTapChangerInterfaceIIDM.cpp
+ *
+ * @brief Ratio tap changer data interface : implementation file for IIDM implementation
+ *
+ */
+
+#include "DYNRatioTapChangerInterfaceIIDM.h"
+
+#include "DYNErrorQueue.h"
+#include "DYNMacrosMessage.h"
+#include "DYNStepInterfaceIIDM.h"
+// #include "DYNTrace.h"
+
+using boost::shared_ptr;
+
+namespace DYN {
+
+RatioTapChangerInterfaceIIDM::~RatioTapChangerInterfaceIIDM() {
+}
+
+RatioTapChangerInterfaceIIDM::RatioTapChangerInterfaceIIDM(powsybl::iidm::RatioTapChanger& tapChanger, const std::string& parentName) : tapChangerIIDM_(tapChanger) {
+  sanityCheck(parentName);
+}
+
+void
+RatioTapChangerInterfaceIIDM::addStep(const shared_ptr<StepInterface>& step) {
+  steps_.push_back(step);
+}
+
+std::vector<shared_ptr<StepInterface> >
+RatioTapChangerInterfaceIIDM::getSteps() const {
+  return steps_;
+}
+
+int
+RatioTapChangerInterfaceIIDM::getCurrentPosition() const {
+  return tapChangerIIDM_.getTapPosition();  // getTapPosition() is 'long' in powsybl
+}
+
+void
+RatioTapChangerInterfaceIIDM::setCurrentPosition(const int& position) {
+  tapChangerIIDM_.setTapPosition(position);
+}
+
+int
+RatioTapChangerInterfaceIIDM::getLowPosition() const {
+  return tapChangerIIDM_.getLowTapPosition();
+}
+
+unsigned int
+RatioTapChangerInterfaceIIDM::getNbTap() const {
+  return steps_.size();
+}
+
+bool
+RatioTapChangerInterfaceIIDM::hasLoadTapChangingCapabilities() const {
+  return tapChangerIIDM_.hasLoadTapChangingCapabilities();
+}
+
+bool
+RatioTapChangerInterfaceIIDM::getRegulating() const {
+  return tapChangerIIDM_.isRegulating();
+}
+
+double
+RatioTapChangerInterfaceIIDM::getTargetV() const {
+  if (std::isnan(tapChangerIIDM_.getTargetV())) {
+    return 99999.9;
```

Comment:
Please do not modify the value by default returned by this method

Reply:
done :)